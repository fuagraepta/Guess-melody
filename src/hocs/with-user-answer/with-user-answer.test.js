import React from 'react';
import renderer from 'react-test-renderer';

import withUserAnser from './with-user-answer.js';

const MockComponent = () => {
  return (
    <div></div>
  );
};

const mockQuestion = {
  genre: `rock`,
  answers: [
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`,
      genre: `rock`
    },
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`,
      genre: `pop`
    },
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`,
      genre: `jazz`
    },
    {
      src: `https://upload.wikimedia.org/wikipedia/commons/6/64/Ugandan_national_anthem%2C_performed_by_the_U.S._Navy_Band.ogg`,
      genre: `rock`
    }
  ]
};

const MockComponentWrapped = withUserAnser(MockComponent);

describe(`HOC with user answer is rendered correctly.`, () => {
  it(`Component correctly renders with HOC.`, () => {
    const mockComponentWrapped = renderer
    .create(
        <MockComponentWrapped
          question = {mockQuestion}
          userAnsers = {[]}
          onAnswer = {jest.fn()}
          onChange = {jest.fn()}
        />
    )
    .toJSON();

    expect(mockComponentWrapped).toMatchSnapshot();
  });
});
