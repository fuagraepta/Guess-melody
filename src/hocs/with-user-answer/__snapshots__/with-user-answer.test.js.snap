// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC with user answer is rendered correctly. Component correctly renders with HOC. 1`] = `<div />`;
